homeassistant/components/husqvarna_automower/switch.py

```diff
@@ -92,3 +114,100 @@ async def async_turn_on(self, **kwargs: Any) -> None:
             raise HomeAssistantError(
                 f"Command couldn't be sent to the command queue: {exception}"
             ) from exception
+
+
+class AutomowerStayOutZoneSwitchEntity(AutomowerControlEntity, SwitchEntity):
+    """Defining the Automower stay out zone switch."""
+
+    _attr_translation_key = "stay_out_zones"
+
+    def __init__(
+        self,
+        coordinator: AutomowerDataUpdateCoordinator,
+        mower_id: str,
+        stay_out_zone_uid: str,
+    ) -> None:
+        """Set up Automower switch."""
+        super().__init__(mower_id, coordinator)
+        self.coordinator = coordinator
+        self.stay_out_zone_uid = stay_out_zone_uid
+        self._attr_unique_id = (
+            f"{self.mower_id}_{stay_out_zone_uid}_{self._attr_translation_key}"
+        )
+        self._attr_translation_placeholders = {"stay_out_zone": self.stay_out_zone.name}
+
+    @property
+    def stay_out_zones(self) -> StayOutZones:
+        """Return all stay out zones."""
+        if TYPE_CHECKING:
+            assert self.mower_attributes.stay_out_zones is not None
+        return self.mower_attributes.stay_out_zones
+
+    @property
+    def stay_out_zone(self) -> Zone:
+        """Return the specific stay out zone."""
+        return self.stay_out_zones.zones[self.stay_out_zone_uid]
+
+    @property
+    def is_on(self) -> bool:
+        """Return the state of the switch."""
+        return self.stay_out_zone.enabled
+
+    @property
+    def available(self) -> bool:
+        """Return True if the device is available and the zones are not `dirty`."""
+        return super().available and not self.stay_out_zones.dirty
+
+    async def async_turn_off(self, **kwargs: Any) -> None:
+        """Turn the switch off."""
+        try:
+            await self.coordinator.api.commands.switch_stay_out_zone(
+                self.mower_id, self.stay_out_zone_uid, False
+            )
+            # As there are no updates from the websocket regarding stay out zone changes,
+            # we need to wait 5s and then poll the API.
+            await asyncio.sleep(5)
```

Comment:
Where are the 5 seconds coming from?
We should create a const for it and document it only once

Reply:
I determined the 5s empirically. Constant created in https://github.com/home-assistant/core/pull/117809/commits/6810ac1ccb9a28b08d03c01a2bd81cf0d6c2fcf4